cmd/kots/cli/identity.go

```diff
@@ -95,6 +101,81 @@ func IdentityServiceInstallCmd() *cobra.Command {
 
 	cmd.Flags().String("identity-config", "", "path to a yaml file containing the KOTS identity service configuration")
 
+	// random other registry flags
+	cmd.Flags().String("license-file", "", "path to a license file to use when download a replicated app")
+	cmd.Flags().String("airgap-bundle", "", "path to the application airgap bundle where application metadata will be loaded from")
+	cmd.Flags().Bool("airgap", false, "set to true to run install in airgapped mode. setting --airgap-bundle implies --airgap=true.")
+
+	registryFlags(cmd.Flags())
+
+	return cmd
+}
+
+func IdentityServiceConfigureCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:           "configure",
+		Short:         "Configure the KOTS Identity Service",
+		SilenceUsage:  true,
+		SilenceErrors: false,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			viper.BindPFlags(cmd.Flags())
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			v := viper.GetViper()
+
+			log := logger.NewLogger()
+
+			clientset, err := k8sutil.GetClientset(kubernetesConfigFlags)
+			if err != nil {
+				return errors.Wrap(err, "failed to get clientset")
+			}
+
+			namespace := v.GetString("namespace")
+			if err := validateNamespace(namespace); err != nil {
+				return err
+			}
+
+			ingressConfig, err := ingress.GetConfig(cmd.Context(), namespace)
+			if err != nil {
+				return errors.Wrap(err, "failed to get ingress config")
+			}
+			if !ingressConfig.Enabled {
+				return errors.New("ingress is not enabled")
+			}
+
+			identityConfig := identitytypes.Config{}
+			if identityConfigPath := v.GetString("identity-config"); identityConfigPath != "" {
+				content, err := ioutil.ReadFile(identityConfigPath)
+				if err != nil {
+					return errors.Wrap(err, "failed to read identity service config file")
+				}
+				if err := ghodssyaml.Unmarshal(content, &identityConfig); err != nil {
+					return errors.Wrap(err, "failed to unmarshal identity service config yaml")
+				}
+			}
+
+			log.ChildActionWithSpinner("Configuring the Identity Service")
+
+			identityConfig.Enabled = true
+			identityConfig.DisablePasswordAuth = true
+			identityConfig.IngressConfig.Enabled = true
+
+			if err := identity.SetConfig(cmd.Context(), namespace, identityConfig); err != nil {
+				return errors.Wrap(err, "failed to set identity config")
+			}
+
+			if err := identity.DeploySecret(cmd.Context(), log, clientset, namespace, identityConfig, *ingressConfig); err != nil {
```

Comment:
same here

Reply:
why? that means you will have to include all registry settings for airgap each time you want to make a change to the identity config. is there some reason you need to remove that function?

Reply:
i added the function back